### PR TITLE
Add table_aws_ecr_repository. closes #115

### LINKED ISSUE #115: Add table_aws_ecr_repository

### PR DESCRIPTION
**Pull request by Priyanka585464**
```
SETUP: tests/aws_ecr_repository []

PRETEST: tests/aws_ecr_repository

TEST: tests/aws_ecr_repository
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_ecr_repository.named_test_resource: Creating...
aws_ecr_repository.named_test_resource: Creation complete after 4s [id=turbottest27627]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
registry_id = 986325076436
resource_aka = arn:aws:ecr:us-east-1:986325076436:repository/turbottest27627
resource_id = turbottest27627
resource_name = turbottest27627

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:ecr:us-east-1:986325076436:repository/turbottest27627"
    ],
    "image_tag_mutability": "MUTABLE",
    "partition": "aws",
    "region": "us-east-1",
    "repository_name": "turbottest27627"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:ecr:us-east-1:986325076436:repository/turbottest27627"
    ],
    "repository_name": "turbottest27627",
    "tags": {
      "name": "turbottest27627"
    },
    "title": "turbottest27627"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:ecr:us-east-1:986325076436:repository/turbottest27627",
    "image_scanning_configuration": {
      "ScanOnPush": false
    },
    "image_tag_mutability": "MUTABLE",
    "partition": "aws",
    "region": "us-east-1",
    "repository_name": "turbottest27627"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:ecr:us-east-1:986325076436:repository/turbottest27627"
    ],
    "region": "us-east-1",
    "title": "turbottest27627"
  }
]
✔ PASSED

POSTTEST: tests/aws_ecr_repository

TEARDOWN: tests/aws_ecr_repository

SUMMARY:

1/1 passed.
```

### CODE REVIEW

File: aws/table_aws_ecr_repository.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:12]:
```suggestion

//// TABLE DEFINITION

```

File: aws/table_aws_ecr_repository.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+			Hydrate:           getAwsECRRepositories,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAwsECRRepositories,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "repository_name",
+				Description: "A list of repositories to describe.",
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:29]:
```suggestion
				Description: "The name of the repository.",
```

File: aws/table_aws_ecr_repository.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "max_results",
+				Description: "The maximum number of repository results returned by DescribeRepositories.",
+				Hydrate:     getAwsECRRepositories,
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "next_token",
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:44]: why are you adding this column? This is not inside `Repositories`

File: aws/table_aws_ecr_repository.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+				Description: "The image scanning configuration for a repository.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "image_details",
+				Description: "A list of ImageDetail objects that contain data about the image.",
+				Hydrate:     getAwsECRDescribeImages,
+				Type:        proto.ColumnType_JSON,
+			},
+			/// Standard columns for all tables
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:86]:
```suggestion

			/// Standard columns for all tables
```

File: aws/table_aws_ecr_repository.go [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+				Name:        "image_scanning_configuration",
+				Description: "The image scanning configuration for a repository.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "image_details",
+				Description: "A list of ImageDetail objects that contain data about the image.",
+				Hydrate:     getAwsECRDescribeImages,
+				Type:        proto.ColumnType_JSON,
+			},
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:85]: Add the tags_src column

File: aws/table_aws_ecr_repository.go [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+				d.StreamListItem(ctx, repository)
+
+			}
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+func getAwsECRRepositories(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:140]:
```suggestion

////  HYDRATE FUNCTIONS

func getAwsECRRepositories(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_ecr_repository.go [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+
+	// Get call
+	op, err := svc.ListTagsForResource(params)
+	if err != nil {
+		logger.Debug("getAwsECRRepositoryTags", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:208]:
```suggestion
}

```

File: aws/table_aws_ecr_repository.go [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+
+	// Get call
+	op, err := svc.GetRepositoryPolicy(params)
+	if err != nil {
+		logger.Debug("getRepositoryPolicy", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:240]:
```suggestion
}

```

File: aws/table_aws_ecr_repository.go [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+
+	// Get call
+	op, err := svc.DescribeImages(params)
+	if err != nil {
+		logger.Debug("getAwsECRDescribeImages", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:273]:
```suggestion
}

```

File: docs/tables/aws_ecr_repository.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+	repository_name,
+	registry_id,
+	repository_arn,
+	repository_uri,
+	created_at,
+	region,
+	account_id
+from
+	aws_ecr_repository;
+```
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:20]:
```
select
  repository_name,
  registry_id,
  repository_arn,
  repository_uri,
  created_at,
  region,
  account_id
from
  aws_ecr_repository;
```

Comment (Paulami30) [reply docs/tables/aws_ecr_repository.md:20]: Fix indentation for all the queries.

File: docs/tables/aws_ecr_repository.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+```sql
+select
+	repository_name,
+	encryption_configuration ->> 'EncryptionType' as encryption_type,
+	encryption_configuration ->> 'KmsKey' as kms_key
+from
+	aws_ecr_repository
+where
+	encryption_configuration ->> 'KmsKey' = 'KMS';
+```
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:78]: The below 2 queries should be the 2nd and 3rd query respectively

File: docs/tables/aws_ecr_repository.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+```sql
+select
+	repository_name,
+	image_tag_mutability
+from
+	aws_ecr_repository
+where
+	image_tag_mutability = 'MUTABLE';
+```
+
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:32]:
```suggestion


```

File: docs/tables/aws_ecr_repository.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+```sql
+select
+  repository_name,
+	image_scanning_configuration as scan
+from
+	aws_ecr_repository
+where
+	finding-severity-counts= 'CRITICAL';
+```
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:43]:
```suggestion
```


```

Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:43]:
```suggestion
```


```

File: docs/tables/aws_ecr_repository.md [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+### Count of image ids in a requested Repository
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:44]:
```suggestion


### Count of image ids in a requested Repository
```

File: docs/tables/aws_ecr_repository.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+```sql
+select
+	count(image_ids) as count
+from
+	aws_ecr_repository
+where
+	repository_name = 'steampipe-12';
+```
+### List of unencrypted ECR repository
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:54]:
```suggestion


### List of unencrypted ECR repository
```

File: docs/tables/aws_ecr_repository.md [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+select
+	repository_name,
+	encryption_configuration ->> 'EncryptionType' as encryption_type,
+	encryption_configuration ->> 'KmsKey' as kms_key
+from
+	aws_ecr_repository
+where
+	encryption_configuration ->> 'KmsKey' = 'null';
+```
+
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:66]:
```suggestion


```

File: docs/tables/aws_ecr_repository.md [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+### Count of image ids in a requested Repository
+
+```sql
+select
+	count(image_ids) as count
+from
+	aws_ecr_repository
+where
+	repository_name = 'steampipe-12';
+```
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:53]: The above query is not working. Rewrite the query according to the heading

File: docs/tables/aws_ecr_repository.md [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+### List Repositories who has image scan configuration with both CRITICAL findings 
+
+```sql
+select
+  repository_name,
+	image_scanning_configuration as scan
+from
+	aws_ecr_repository
+where
+	finding-severity-counts= 'CRITICAL';
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:42]: I can not find any column `finding-severity-counts` by this name

File: docs/tables/aws_ecr_repository.md [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+select
+	repository_name,
+	image_tag_mutability
+from
+	aws_ecr_repository
+where
+	image_tag_mutability = 'MUTABLE';
+```
+
+### List Repositories who has image scan configuration with both CRITICAL findings
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:33]: the heading is not appropriate

File: docs/tables/aws_ecr_repository.md [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+	repository_name,
+	encryption_configuration ->> 'EncryptionType' as encryption_type,
+	encryption_configuration ->> 'KmsKey' as kms_key
+from
+	aws_ecr_repository
+where
+	encryption_configuration ->> 'KmsKey' = 'null';
+```
+
+### List of ECR repository which are not using Customer Managed Keys(CMK)
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:67]:
```suggestion
### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption
```

File: aws-test/tests/aws_ecr_repository/variables.tf [hunk 21]
```diff
@@ -0,0 +1,88 @@
+
+
```
Comment (Paulami30) [inline aws-test/tests/aws_ecr_repository/variables.tf:2]: remove space

File: aws-test/tests/aws_ecr_repository/test-turbot-query.sql [hunk 22]
```diff
@@ -0,0 +1,3 @@
+select account_id, akas, region, title
+from aws.aws_ecr_repository
+where repository_name = '{{ output.resource_name.value }}'
```
Comment (Paulami30) [inline aws-test/tests/aws_ecr_repository/test-turbot-query.sql:3]:
```suggestion
where repository_name = '{{ output.resource_name.value }}';
```

File: aws-test/tests/aws_ecr_repository/test-notfound-query.sql [hunk 23]
```diff
@@ -0,0 +1,4 @@
+select registry_id, repository_name, repository_arn, region, account_id
+from aws.aws_ecr_repository
+where repository_name = '{{ output.resourceName }}.1'
```
Comment (Paulami30) [inline aws-test/tests/aws_ecr_repository/test-notfound-query.sql:3]:
```suggestion
where repository_name = '{{ output.resourceName }}.1';
```

File: aws-test/tests/aws_ecr_repository/test-list-query.sql [hunk 24]
```diff
@@ -0,0 +1,3 @@
+select repository_name, repository_arn, image_tag_mutability, image_scanning_configuration, partition, region
+from aws.aws_ecr_repository
+where akas::text = '["{{ output.resource_aka.value }}"]'
```
Comment (Paulami30) [inline aws-test/tests/aws_ecr_repository/test-list-query.sql:3]:
```suggestion
where akas::text = '["{{ output.resource_aka.value }}"]';
```
Please add ; for all the sql queries

**Review by Paulami30 — COMMENTED** [review]
Please make the suggested changes

File: docs/tables/aws_ecr_repository.md [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+  repository_arn,
+  repository_uri,
+  created_at,
+  region,
+  account_id
+from
+  aws_ecr_repository;
+```
+
+### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:21]:
```suggestion
### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption

```

File: aws/table_aws_ecr_repository.go [hunk 26]
```diff
@@ -0,0 +1,333 @@
...
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the Repository",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsECRRepositoryTags,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			/// Standard columns for all tables
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:90]:
```suggestion
			// Standard columns for all tables
```

File: docs/tables/aws_ecr_repository.md [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+  repository_name,
+  encryption_configuration ->> 'EncryptionType' as encryption_type,
+  encryption_configuration ->> 'KmsKey' as kms_key
+from
+  aws_ecr_repository
+where
+  encryption_configuration ->> 'KmsKey' = 'KMS';
+```
+
+### List Repositories whose imageTag set as MUTABLE
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:33]:
```suggestion
### List Repositories whose imageTag set as MUTABLE

```
Add this space in all the query

File: docs/tables/aws_ecr_repository.md [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+
+### Count of image details in each Repository
+```sql
+select
+  count(image_details) as count
+from
+  aws_ecr_repository
+where
+  repository_name = 'steampipe-12';
+```
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:52]: Is the above query working? `image_details` is an array

File: docs/tables/aws_ecr_repository.md [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+```sql
+select
+  repository_name,
+  image_tag_mutability
+from
+  aws_ecr_repository
+where
+  image_tag_mutability = 'MUTABLE';
+```
+
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:43]: Add a new query `List of repositories where automatic image scanning is disable`

File: docs/tables/aws_ecr_repository.md (hunk 28 above)
Comment (Priyanka585464) [reply docs/tables/aws_ecr_repository.md:52]: yes it's working.

File: docs/tables/aws_ecr_repository.md [hunk 30]
```diff
@@ -0,0 +1,88 @@
...
+  image_scanning_configuration ->> 'ScanOnPush' = 'false';
+```
+
+
+### Image vulnerability count by severity. 
+
+```sql
+select
+  repository_name,
+  detail -> 'ImageScanFindingsSummary' -> 'FindingSeverityCounts' -> 'INFORMATIONAL' as severity_counts
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:68]:
```suggestion
  detail -> 'ImageScanFindingsSummary' -> 'FindingSeverityCounts' -> 'INFORMATIONAL' as informational_severity_counts
```

Comment (Paulami30) [reply docs/tables/aws_ecr_repository.md:68]: show the low and the medium severity count too

**Review by Paulami30 — APPROVED** [review]

**Review by Paulami30 — COMMENTED** [review]
resolve merge conflict

File: aws/service.go [hunk 31]
```diff
@@ -233,6 +234,16 @@ func Ec2Service(ctx context.Context, d *plugin.QueryData, region string) (*ec2.E
...
 
+// ECRService returns the service connection for AWS ECR service
+func EcrService(ctx context.Context, d *plugin.QueryData, region string) (*ecr.ECR, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed EcrService")
+	}
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("ecr-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*ecr.ECR), nil
```
Comment (khushboo9024) [inline aws/service.go:245]: Merged conflict was bot resolved properly, this code is not going to work.

File: aws/table_aws_ecr_repository.go [hunk 32]
```diff
@@ -0,0 +1,333 @@
...
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns for all tables
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsECRRepositoryTags,
+				Transform:   transform.FromField("Tags"),
```
Comment (khushboo9024) [inline aws/table_aws_ecr_repository.go:96]:
```suggestion
				Transform:   transform.FromField("Tags").transform.From(ecrTagListToTurbotTags),
```

**Comment by cbruno10** [comment]
@Priyanka585464 Can you please see the review comments when you get a chance? Thanks!

File: aws/service.go [hunk 33]
```diff
@@ -272,6 +283,7 @@ func EcsService(ctx context.Context, d *plugin.QueryData, region string) (*ecs.E
 	if err != nil {
 		return nil, err
 	}
+	svc := ecr.New(sess)
```
[the remark this answers is not in the document]
Comment (Priyanka585464) [reply aws/service.go]: There was some conflicts in service conn.

File: aws/plugin.go [hunk 34]
```diff
@@ -69,6 +69,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_ec2_transit_gateway":                tableAwsEc2TransitGateway(ctx),
 			"aws_ec2_transit_gateway_route_table":    tableAwsEc2TransitGatewayRouteTable(ctx),
 			"aws_ec2_transit_gateway_vpc_attachment": tableAwsEc2TransitGatewayVpcAttachment(ctx),
+			"aws_ecr_repository":                     tableAwsECRRepository(ctx),
 			"aws_ecs_cluster":                        tableAwsEcsCluster(ctx),
     	"aws_ecs_task_definition":                tableAwsEcsTaskDefinition(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:74]: fix indentation

File: docs/tables/aws_ecr_repository.md [hunk 35]
```diff
@@ -0,0 +1,92 @@
...
+  repository_name,
+  image_tag_mutability
+from
+  aws_ecr_repository
+where
+  image_tag_mutability = 'IMMUTABLE';
+```
+
+
+### Repository image vulnerability count by severity
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:78]:
```suggestion
### Get repository image vulnerability count by severity 
```

File: aws/table_aws_ecr_repository.go [hunk 36]
```diff
@@ -0,0 +1,334 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("RepositoryArn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsECRRepositories(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:117]:
```suggestion
func listAwsEcrRepositories(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_ecr_repository.go [hunk 37]
```diff
@@ -0,0 +1,334 @@
...
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+////  HYDRATE FUNCTIONS
+
+func getAwsECRRepositories(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:149]:
```suggestion
func getAwsEcrRepositories(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_ecr_repository.go [hunk 38]
```diff
@@ -0,0 +1,334 @@
...
+	// Get call
+	op, err := svc.ListTagsForResource(params)
+	if err != nil {
+		logger.Debug("getAwsECRRepositoryTags", "ERROR", err)
+		return nil, err
+	}
+	return op, nil
+}
+
+func getRepositoryPolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:218]:
```suggestion
func getAwsEcrRepositoryPolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_ecr_repository.go [hunk 39]
```diff
@@ -0,0 +1,334 @@
...
+	op, err := svc.GetRepositoryPolicy(params)
+	if err != nil {
+		logger.Debug("getRepositoryPolicy", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func getAwsECRDescribeImages(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:251]:
```suggestion
func getAwsEcrDescribeImages(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_ecr_repository.go [hunk 40]
```diff
@@ -0,0 +1,334 @@
...
+	op, err := svc.DescribeImages(params)
+	if err != nil {
+		logger.Debug("getAwsECRDescribeImages", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func getLifecyclePolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_ecr_repository.go:285]:
```suggestion
func getAwsEcrRepositoryLifecyclePolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: docs/tables/aws_ecr_repository.md [hunk 41]
```diff
@@ -0,0 +1,90 @@
+# Table: aws_ecr_repository
+
+Amazon Elastic Container Registry (Amazon ECR) is a managed container image registry service.
+
+## Examples
+
+### ECR repository basic info
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:7]:
```suggestion
### Basic info
```

File: docs/tables/aws_ecr_repository.md [hunk 42]
```diff
@@ -0,0 +1,90 @@
...
+  encryption_configuration ->> 'EncryptionType' as encryption_type,
+  encryption_configuration ->> 'KmsKey' as kms_key
+from
+  aws_ecr_repository
+where
+  encryption_configuration ->> 'EncryptionType' = 'AES256';
+```
+
+
+### List repositories where automatic image scanning is disabled
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:37]:
```suggestion
### List repositories whose automatic image scanning is disabled
```

**Review by Paulami30 — COMMENTED** [review]
Please check the comments, resolve merge conflict and run the integration test and update the test log. Thanks

**Review by Paulami30 — APPROVED** [review]

**Comment by Paulami30** [comment]
LGTM

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
I see you are pulling the policies (getAwsEcrRepositoryPolicy), but I am not seeing any col for it. Assume that we would like to check that ECR repositories are not exposed to everyone, can we check that at this stage ?

**Comment by rajmohanty17** [comment]
We are already the API calls for this 
https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_GetRepositoryPolicy.html 
and https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_GetLifecyclePolicy.html

Add the respective columns to evaluate some use cases such as (example)

ECR repositories are using lifecycle policies for cost optimization.
ECR repositories are not exposed to everyone.

File: docs/tables/aws_ecr_repository.md [hunk 43]
```diff
@@ -0,0 +1,126 @@
...
+  repository_name,
+  r -> 'selection' ->> 'tagStatus' as tag_status,
+  r -> 'selection' ->> 'countType' as count_type
+from
+  aws_ecr_repository,
+  jsonb_array_elements(lifecycle_policy -> 'rules') as r
+where
+  ((r -> 'selection' ->> 'tagStatus' <>'untagged')
+  and (r -> 'selection' ->> 'countType' <>'sinceImagePushed'));
+```
```
Comment (Paulami30) [inline docs/tables/aws_ecr_repository.md:91]: format the query

**Review by Paulami30 — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM for now. Good to see 2 new columns.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**